sdk/python/feast/errors.py

```diff
@@ -78,6 +78,16 @@ def __init__(self, offline_store_name: str, data_source_name: str):
         )
 
 
+class FeatureNameCollisionError(Exception):
+    def __init__(self, feature_refs_collisions: List[str]):
+        feature_name_collisions = [ref.split(":")[1] for ref in feature_refs_collisions]
+        feature_names = ", ".join(x for x in feature_name_collisions)
+        super().__init__(
+            f"The following feature name(s) have collisions: {feature_names}. Set 'full_feature_names' "
```

Comment:
The option provided to the user is not the only way to solve the problem. Ensuring that the features have different names is another option.

Reply:
I have made the error message much more descriptive, hopefully that clears things up. For example, now a list of feature refs with 2 feature refs `driver_hourly_stats:conv_rate` would return the error message below. This generalizes across multiple feature views also.

```
The following feature name(s) have collisions: conv_rate. Set 'full_feature_names' argument in the data retrieval function to True to use the full feature name which is prefixed by the feature view name, or rename colliding features.
Collisions occur in the following feature view(s): driver_hourly_stats.
```

Reply:
Thanks @Mwad22, going to be nitpicky now.

I took a stab at the error message for collisions where prefixes are missing and the name is the same.
```
Duplicate features named conv_rate found.

To resolve this collision, either use the full feature name by setting 'full_feature_names=True', or ensure that the features in question have different names.
```
or when prefixes are already in use

```
Duplicate features named driver_conv_rate found.

To resolve this collision, please ensure that the features in question have different names.
```

Reply:
So, for the case where feature names are not prefixed, we no longer want to mention the feature views where the collisions occur?

Reply:
@woop I went ahead and pushed the change. Feel free to ignore my question above, as I think the error messages are descriptive enough in the cases you describe above.